website/src/components/Edit/InputForm.tsx

```diff
@@ -0,0 +1,271 @@
+import type { InputField } from '../../types/config';
+
+type SubtitleProps = {
+    title: string;
+    bold?: boolean;
+    small?: boolean;
+    customKey?: string;
+};
+export const Subtitle: FC<SubtitleProps> = ({ title, bold, small, customKey }) => (
+    <Fragment key={snakeCase(customKey ?? title) + '_fragment'}>
+        <tr key={snakeCase(customKey ?? title) + '_spacing'} className='h-4' />
+        <tr key={snakeCase(customKey ?? title)} className='subtitle'>
+            <td className={`${(bold ?? false) ? 'font-semibold' : 'font-normal'} ${small && 'text-base'}`} colSpan={3}>
+                {title}
+            </td>
+        </tr>
+    </Fragment>
+);
+
+/**
+ * Immutable class used by the {@link MetadataForm}. 'Mutate' objects with the {@link updateWith} method.
+ * Also encapsulates functionality to turn the metadata either into a TSV file or into a Record for API
+ * submission.
+ */
+export class EditableMetadata {
+    private constructor(public readonly rows: Row[]) {}
+
+    static fromInitialData(initialData: SequenceEntryToEdit): EditableMetadata {
+        return new EditableMetadata(
+            Object.entries(initialData.originalData.metadata).map(([key, value]) => ({
+                key,
+                value,
+                initialValue: value,
+                ...mapErrorsAndWarnings(initialData, key, 'Metadata'),
+            })),
+        );
+    }
+
+    static empty(): EditableMetadata {
+        return new EditableMetadata([]);
+    }
+
+    updateWith(editedRow: Row): EditableMetadata {
+        const relevantOldRow = this.rows.find((oldRow) => oldRow.key === editedRow.key);
+        return new EditableMetadata(
+            relevantOldRow
+                ? this.rows.map((prevRow) =>
+                      prevRow.key === editedRow.key ? { ...prevRow, value: editedRow.value } : prevRow,
+                  )
+                : [...this.rows, editedRow],
+        );
+    }
+
+    /**
+     * Helper function to get the Submission ID from the rows, if it is present.
+     */
+    getSubmissionId(): string | undefined {
+        const row = this.rows.find((row) => row.key === SUBMISSION_ID_FIELD);
+        return row ? row.value : undefined;
+    }
+
+    /**
+     * Return the Metadata information as a TSV. If no information is present, 'undefined' is returned.
+     * @param submissionId optional (might already be in the rows if add to the form initially).
+     *      The submission ID to put into the TSV.
+     * @param accession optional. If an accession is already assigned to this sequence, it should be given.
+     */
+    getMetadataTsv(submissionId?: string, accession?: string): File | undefined {
+        // if no values are set at all, return undefined
+        if (!this.rows.some((row) => row.value !== '')) return undefined;
+
+        // If only the submission ID is set, that't not enough, also return undefined.
```

Comment:
ah this relates to my comment above - not sure why we need this?

Reply:
I submitted only the submission ID before, and then an error came back from the backend - something like "metadata file does not contain metadata for submissionID X" and I wanted to avoid this error popping up because it mentions a file that the user doesn't know about. So it seemed like the easiest way to prevent this was to just do a check beforehand.

Reply:
Makes sense! We can make a new issue to allow sequences without metadata

Reply:
I'm not entirely sure what to put into such an issue - I'll leave it up to you to maybe create one!